languages/elixir/exercises/concept/maps/.docs/introduction.md

```diff
@@ -0,0 +1,93 @@
+Maps in Elixir are the goto data structure for storing information in key-value pairs. In other languages, these might also be known as associative arrays (PHP), hashes (Perl 5, Raku), or dictionaries (Python).
+
+Keys and values can be of any data type, but if the key is an atom we can make use of special access protocols. Maps do not guarantee the order of their entries when access or returned.
+
+> Elixir gotcha! When an instance of a map has a small number of entries they appear to be ordered because of their internal implementation as an ordered keyword list, when their size exceeds 32 items, it is then stored as a [hash array mapped tree][hamt].
+
+## Literal forms
+
+An empty map is simply declared with `%{}`. If we want to add items to a map literal, we can use two forms:
+
+```elixir
+# If the key is an atom:
+%{atom_key: 1}
+
+# If the key is a different type:
+%{1 => :atom_value}
+
+# You can even mix these if the atom form comes second:
+%{"first_form" => :a, atom_form: :b}
+```
+
+It would be convention to choose a consistent form which first all of the data in the literal.
+
+## Map module functions
+
+Elixir provides many functions for working with maps to access/get/put/remove/drop values and keys.
+
+```shell
+iex> map = %{}
+%{}
+iex> map_a = Map.put(map, :a, 1)
+%{a: 1}
+iex> Map.get(map_a, :a)
+1
+```
+
+Some functions require an anonymous or captured function to be passed into the function:
```

Comment:
As one of the prerequisites is `anonymous-functions`, we can expect students to already know about anonymous functions, right? If so, I think this bit of text can be removed.